bitcoin/src/psbt/mod.rs

```diff
@@ -56,9 +142,161 @@ pub struct Psbt {
     pub inputs: Vec<Input>,
     /// The corresponding key-value map for each output in the unsigned transaction.
     pub outputs: Vec<Output>,
+
+    // PsbtV2 fields
+    /// 32-bit little endian signed integer representing the
+    /// version number of the transaction being created
+    pub tx_version: Option<i32>,
+    /// 32-bit little endian unsigned integer representing the transaction locktime
+    /// to use if no inputs specify a required locktime.
+    pub fallback_locktime: Option<absolute::LockTime>,
+    /// PSBTv2 field for various transaction modification flags
+    pub tx_modifiable: Option<TxModifiable>,
+}
+
+impl PsbtInner {
+    /// Validates the inputs according to the [`PsbtInner`] version
+    pub fn validate_inputs_version(&self) -> Result<(), Error> {
```

Comment:
Honestly, I was not sure if making these functions public would help developers in any way, so added them in case they did. Maybe it is better to just inline them into `Psbt::validate`.